libafl_bolts/src/tuples.rs

```diff
@@ -731,6 +731,43 @@ where
     }
 }
 
+/// Trait for structs which are capable of mapping a given type to another.
+pub trait Mapper<T> {
+    /// The result of the mapping operation.
+    type Output;
+
+    /// The actual mapping operation.
+    fn map(&mut self, from: T) -> Self::Output;
```

Comment:
It's a bit confusing to use `map` since we have ShMap etc. a lot in the codebase

Reply:
Named after: https://doc.rust-lang.org/nightly/core/iter/struct.Map.html

Reply:
Fair point, I guess